thingif/src/main/java/com/kii/thingif/internal/gson/PredicateAdapter.java

```diff
@@ -0,0 +1,81 @@
+package com.kii.thingif.internal.gson;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import com.kii.thingif.clause.trigger.TriggerClause;
+import com.kii.thingif.trigger.Condition;
+import com.kii.thingif.trigger.Predicate;
+import com.kii.thingif.trigger.ScheduleOncePredicate;
+import com.kii.thingif.trigger.SchedulePredicate;
+import com.kii.thingif.trigger.StatePredicate;
+import com.kii.thingif.trigger.TriggersWhen;
+
+import java.lang.reflect.Type;
+
+public class PredicateAdapter implements
+        JsonSerializer<Predicate>,
+        JsonDeserializer<Predicate>{
+    private Gson gson = new GsonBuilder()
+            .registerTypeAdapter(TriggerClause.class, new TriggerClauseAdapter())
+            .create();
+    @Override
+    public JsonElement serialize(Predicate src, Type typeOfSrc, JsonSerializationContext context) {
+        if (src == null) return null;
+
+        if (src instanceof StatePredicate) {
+            JsonElement clause = gson.toJsonTree(
+                    ((StatePredicate) src).getCondition().getClause(),
+                    TriggerClause.class);
+            JsonObject json = new JsonObject();
```

Comment:
This is trivial cosmetic issue but could you rename `json` to `ret`? In `else` clause, return value name is `ret`.